pkg/spoke/addon/configuration.go

```diff
@@ -0,0 +1,222 @@
+package addon
+
+import (
+	"crypto/sha256"
+	"crypto/x509/pkix"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	operatorhelpers "github.com/openshift/library-go/pkg/operator/v1helpers"
+	certificatesv1beta1 "k8s.io/api/certificates/v1beta1"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+const (
+	installationNamespaceAnnotation = "installNamespace"
+	bootstrapSecretAnnotation       = "bootstrapSecret"
+	registrationAnnotation          = "registrations"
+	clusterNamePlaceHolder          = "{ClusterName}"
+	agentNamePlaceHolder            = "{AgentName}"
+)
+
+// registrationConfig contains necessary information for addon registration. A addon may have multiple
+// registrationConfigs.
+type registrationConfig struct {
+	AddonName             string      `json:"addonName"`
+	InstallationNamespace string      `json:"installNamespace"`
+	BootstrapSecretName   string      `json:"bootstrapSecretName,omitempty"`
+	SignerName            string      `json:"signerName,omitempty"`
```

Comment:
i think we will support multiple signers?

It is better to support that in API at the very beginning, and add a todo to implement later.